src/app/shared/auth/auth-info.spec.ts

```diff
@@ -1,72 +1,86 @@
+  clearTokenFromStorage,
+  forcedTokenAuthFromStorage,
+  hasForcedToken,
+  setForcedTokenInStorage,
+  tokenAuthFromServiceResp,
+  tokenAuthFromStorage
+} from './auth-info';
 
 describe('auth-info', () => {
 
+  const expectedToken = 'atoken';
   const expiresInSec = 60;
   const expiresInMS = expiresInSec*SECONDS;
   const precision = -3; // the 3 lower digits may be different = 1s precison
 
   beforeEach(function() {
     clearTokenFromStorage();
+    clearForcedTokenFromStorage();
   });
 
   it('should save & load successfully a token', () => {
     const now = Date.now();
 
-    const auth = tokenAuthFromServiceResp('atemptok', expiresInSec);
+    const auth = tokenAuthFromServiceResp(expectedToken, expiresInSec);
     const loaded = tokenAuthFromStorage();
 
-    expect(auth.accessToken).toEqual('atemptok');
+    expect(auth.accessToken).toEqual(expectedToken);
     expect(auth.creation.getTime()-now).toBeCloseTo(0,precision);
     expect(auth.expiration.getTime()-now).toBeCloseTo(expiresInMS,precision);
 
-    expect(loaded?.accessToken).toEqual('atemptok');
-    if (loaded) {
-      expect(loaded.creation.getTime()).toBeCloseTo(now,precision);
-      expect(loaded.expiration.getTime()).toBeCloseTo(now+expiresInMS,precision);
-    }
+    expect(loaded.accessToken).toEqual(expectedToken);
+    expect(loaded.creation.getTime()).toBeCloseTo(now,precision);
+    expect(loaded.expiration.getTime()).toBeCloseTo(now+expiresInMS,precision);
   });
 
   it('should fail when loading a token with no token', () => {
-    tokenAuthFromServiceResp('atemptok', expiresInSec);
+    tokenAuthFromServiceResp(expectedToken, expiresInSec);
     sessionStorage.removeItem('access_token');
-    const loaded = tokenAuthFromStorage();
-    expect(loaded).toBeUndefined();
+    expect(tokenAuthFromStorage).toThrowError();
   });
 
   it('should fail when loading a token with no expiration', () => {
-    tokenAuthFromServiceResp('atemptok', expiresInSec);
+    tokenAuthFromServiceResp(expectedToken, expiresInSec);
     sessionStorage.removeItem('access_token_exp');
-    const loaded = tokenAuthFromStorage();
-    expect(loaded).toBeUndefined();
+    expect(tokenAuthFromStorage).toThrowError();
   });
 
   it('should fail (and clear) when loading a expired token', () => {
-    tokenAuthFromServiceResp('atemptok', -1 /* expired 1sec ago */);
-    const loaded = tokenAuthFromStorage();
-    expect(loaded).toBeUndefined();
+    tokenAuthFromServiceResp(expectedToken, -1 /* expired 1sec ago */);
+    expect(tokenAuthFromStorage).toThrowError();
     expect(sessionStorage.getItem('access_token')).toBeNull();
     expect(sessionStorage.getItem('access_token_exp')).toBeNull();
   });
 
   it('should fail (and clear) when loading an invalid token', () => {
-    tokenAuthFromServiceResp('atemptok', -1 /* expired 1sec ago */);
+    tokenAuthFromServiceResp(expectedToken, -1 /* expired 1sec ago */);
     sessionStorage.setItem('access_token_exp', 'not a number');
```

Comment:
Without this line, the test would still produce the same result as the test just upper, it might cause maintenance problems later.
If I'm getting it right, the idea would be to fail because access_token_exp is not a number without considering the expiration. If so, changing the expired value to a passing one will do.

Reply:
👍  I think that was error, indeed